src/Microsoft.AspNet.Server.Kestrel/Filter/LibuvStream.cs

```diff
@@ -85,7 +85,7 @@ public override void Flush()
 
         private Task<int> ReadAsync(ArraySegment<byte> buffer)
         {
-            return _input.ReadAsync(buffer);
+            return _input.ReadAsync(buffer.Array, buffer.Offset, buffer.Count);
```

Comment:
Can't create an `ArraySegement` with a null buffer + specified length [or it will throw an exception](https://github.com/dotnet/coreclr/blob/bc146608854d1db9cdbcc0b08029a87754e12b49/src/mscorlib/src/System/ArraySegment.cs#L45).

However a consumption limit needs to be passed through to `CopyTo` else it will consume the whole input rather than just the input related to the current request. Is a bit of abstraction leakiness to make the code reusable; rather than copy paste repeated...